### PR TITLE
Use kubecli API instead of k8s client API

### PR DESCRIPTION
**Pull request by zedi-pramodh**
During PR reviews of code merge from POC branch to master we moved away from kubecli  API and started using k8s client API. But that has caused regressions and also does not support all the features that come from kubevirt.

Moving back to using kubecli API.
Also fixed couple of bugs found during testing. GetVolumeFormat was missing check for PVC type.

Was able to start apps of all types.

1) VMs 2) Container as VM and 3) Native container

sshd-app-native-cont-0b01b               1/1     Running   0          24m

cf4b1766-574b-4974-bfac-be568107225c:/# kubectl get vmi -A
NAMESPACE      NAME                 AGE   PHASE     IP    NODENAME                 READY
eve-kube-app   ubuntu-vm-a0165      40m   Running         hp-server-bottom-shelf   True
eve-kube-app   sshd-vm-cont-c92a1   36m   Running         hp-server-bottom-shelf   True

### CODE REVIEW

File: pkg/pillar/kubeapi/kubeapi.go [hunk 1]
```diff
@@ -92,6 +90,7 @@ func GetNetClientSet() (*netclientset.Clientset, error) {
 	return nclientset, nil
 }
 
+/*
```
Comment (eriknordmark) [inline pkg/pillar/kubeapi/kubeapi.go:93]: Is there no better way to keep this around but not compile it?
And please add a comment about it being removed (I'm guessing you want to keep if it we switch back again in the future.)

Comment (zedi-pramodh) [reply pkg/pillar/kubeapi/kubeapi.go:93]: Yes I will add the comment, yes wanted to keep for reference and switchback if need be.

Comment (zedi-pramodh) [reply pkg/pillar/kubeapi/kubeapi.go]: Done and updated commit.

File: pkg/pillar/go.mod [hunk 2]
```diff
@@ -206,3 +215,35 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace (
+	github.com/lf-edge/eve/pkg/kube/cnirpc => ../kube/cnirpc
```
Comment (milan-zededa) [inline pkg/pillar/go.mod:220]: Do not bring back replace for cnirpc. Ever since it has been merged we do not need that anymore.

Comment (eriknordmark) [reply pkg/pillar/go.mod:220]: Yeah this must be fixed.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
As Milan said, do not bring back the cnirpc replace directive

Are the other replace required (due to the way they have structured their packages?)

**Comment by milan-zededa** [comment]
Btw. I also tested it and can confirm that all types of apps do work now.

**Pull request merged**